In [2]:
import pandas as pd
import sqlite3
from pandasql import sqldf

In [13]:
conn = sqlite3.connect('/Users/carlamoestafa/Downloads/database.sqlite')

In [14]:
query="""
SELECT * FROM sqlite_master;
"""

In [15]:
df_schema = pd.read_sql_query(query, conn)
df_schema.tbl_name.unique()

array(['sqlite_sequence', 'Player_Attributes', 'Player', 'Match',
       'League', 'Country', 'Team', 'Team_Attributes'], dtype=object)

## QUESTION 1 ANSWER

In [41]:
pd.set_option('display.max_columns', None)

In [57]:
query="""
SELECT * FROM Match;
"""
match = pd.read_sql_query(query, conn)
match.head()

id  country_id  league_id     season  stage                 date  \
0   1           1          1  2008/2009      1  2008-08-17 00:00:00   
1   2           1          1  2008/2009      1  2008-08-16 00:00:00   
2   3           1          1  2008/2009      1  2008-08-16 00:00:00   
3   4           1          1  2008/2009      1  2008-08-17 00:00:00   
4   5           1          1  2008/2009      1  2008-08-16 00:00:00   

   match_api_id  home_team_api_id  away_team_api_id  home_team_goal  \
0        492473              9987              9993               1   
1        492474             10000              9994               0   
2        492475              9984              8635               0   
3        492476              9991              9998               5   
4        492477              7947              9985               1   

   away_team_goal  home_player_X1  home_player_X2  home_player_X3  \
0               1             NaN             NaN             NaN   
1               0             NaN             NaN             NaN   
2               3             NaN             NaN             NaN   
3               0             NaN             NaN             NaN   
4               3             NaN             NaN             NaN   

   home_player_X4  home_player_X5  home_player_X6  home_player_X7  \
0             NaN             NaN             NaN             NaN   
1             NaN             NaN             NaN             NaN   
2             NaN             NaN             NaN             NaN   
3             NaN             NaN             NaN             NaN   
4             NaN             NaN             NaN             NaN   

   home_player_X8  home_player_X9  home_player_X10  home_player_X11  \
0             NaN             NaN              NaN              NaN   
1             NaN             NaN              NaN              NaN   
2             NaN             NaN              NaN              NaN   
3             NaN             NaN              NaN              NaN   
4             NaN             NaN              NaN              NaN   

   away_player_X1  away_player_X2  away_player_X3  away_player_X4  \
0             NaN             NaN             NaN             NaN   
1             NaN             NaN             NaN             NaN   
2             NaN             NaN             NaN             NaN   
3             NaN             NaN             NaN             NaN   
4             NaN             NaN             NaN             NaN   

   away_player_X5  away_player_X6  away_player_X7  away_player_X8  \
0             NaN             NaN             NaN             NaN   
1             NaN             NaN             NaN             NaN   
2             NaN             NaN             NaN             NaN   
3             NaN             NaN             NaN             NaN   
4             NaN             NaN             NaN             NaN   

   away_player_X9  away_player_X10  away_player_X11  home_player_Y1  \
0             NaN              NaN              NaN             NaN   
1             NaN              NaN              NaN             NaN   
2             NaN              NaN              NaN             NaN   
3             NaN              NaN              NaN             NaN   
4             NaN              NaN              NaN             NaN   

   home_player_Y2  home_player_Y3  home_player_Y4  home_player_Y5  \
0             NaN             NaN             NaN             NaN   
1             NaN             NaN             NaN             NaN   
2             NaN             NaN             NaN             NaN   
3             NaN             NaN             NaN             NaN   
4             NaN             NaN             NaN             NaN   

   home_player_Y6  home_player_Y7  home_player_Y8  home_player_Y9  \
0             NaN             NaN             NaN             NaN   
1             NaN             NaN             NaN             NaN   
2            

In [43]:
print(match.columns.tolist())

['id', 'country_id', 'league_id', 'season', 'stage', 'date', 'match_api_id', 'home_team_api_id', 'away_team_api_id', 'home_team_goal', 'away_team_goal', 'home_player_X1', 'home_player_X2', 'home_player_X3', 'home_player_X4', 'home_player_X5', 'home_player_X6', 'home_player_X7', 'home_player_X8', 'home_player_X9', 'home_player_X10', 'home_player_X11', 'away_player_X1', 'away_player_X2', 'away_player_X3', 'away_player_X4', 'away_player_X5', 'away_player_X6', 'away_player_X7', 'away_player_X8', 'away_player_X9', 'away_player_X10', 'away_player_X11', 'home_player_Y1', 'home_player_Y2', 'home_player_Y3', 'home_player_Y4', 'home_player_Y5', 'home_player_Y6', 'home_player_Y7', 'home_player_Y8', 'home_player_Y9', 'home_player_Y10', 'home_player_Y11', 'away_player_Y1', 'away_player_Y2', 'away_player_Y3', 'away_player_Y4', 'away_player_Y5', 'away_player_Y6', 'away_player_Y7', 'away_player_Y8', 'away_player_Y9', 'away_player_Y10', 'away_player_Y11', 'home_player_1', 'home_player_2', 'home_player_

In [62]:
query="""
SELECT match_api_id,home_team_api_id,
SUM(home_team_goal) AS total_home_goal
FROM Match
GROUP BY home_team_api_id
ORDER BY total_home_goal DESC;
"""
home_goals = pd.read_sql_query(query, conn)
home_goals.head(1)

match_api_id  home_team_api_id  total_home_goal
0        530239              8633              505

Team who scored the most points when playing at home is team 8633

## QUESTION 2 ANSWER

In [65]:
query="""
SELECT match_api_id,away_team_api_id,
SUM(away_team_goal) AS total_away_goal
FROM Match
GROUP BY away_team_api_id
ORDER BY total_away_goal DESC;
"""
away_goals = pd.read_sql_query(query, conn)
away_goals.head(1)

match_api_id  away_team_api_id  total_away_goal
0        530086              8634              354

No, the team who scored the most points playing away is a different team.

## QUESTION 3 ANSWER

In [77]:
query="""
SELECT match_api_id,home_team_goal, away_team_goal,
COUNT(DISTINCT match_api_id) AS tie_match
FROM Match
GROUP BY match_api_id
HAVING home_team_goal == away_team_goal
;
"""
tie_matches = pd.read_sql_query(query, conn)
tie_matches

match_api_id  home_team_goal  away_team_goal  tie_match
0           483135               0               0          1
1           483136               4               4          1
2           483142               0               0          1
3           483144               1               1          1
4           483154               1               1          1
...            ...             ...             ...        ...
6591       2216657               1               1          1
6592       2216661               3               3          1
6593       2216663               1               1          1
6594       2216669               2               2          1
6595       2216672               2               2          1

[6596 rows x 4 columns]

In [91]:
tie_matches.sum()

match_api_id      483135
home_team_goal         0
away_team_goal         0
tie_match              1
SUM(tie_match)      6596
dtype: int64

In [95]:
query="""
SELECT SUM(tie_match) FROM
(SELECT match_api_id,home_team_goal, away_team_goal,
COUNT(DISTINCT match_api_id) AS tie_match
FROM Match
GROUP BY match_api_id
HAVING home_team_goal == away_team_goal) AS total_tie_matches;
"""
tie_matches = pd.read_sql_query(query, conn)
tie_matches

SUM(tie_match)
0            6596

## QUESTION 4 ANSWER

In [96]:
query="""
SELECT * FROM Player;
"""
players = pd.read_sql_query(query, conn)
players.head()

id  player_api_id         player_name  player_fifa_api_id  \
0   1         505942  Aaron Appindangoye              218353   
1   2         155782     Aaron Cresswell              189615   
2   3         162549         Aaron Doran              186170   
3   4          30572       Aaron Galindo              140161   
4   5          23780        Aaron Hughes               17725   

              birthday  height  weight  
0  1992-02-29 00:00:00  182.88     187  
1  1989-12-15 00:00:00  170.18     146  
2  1991-05-13 00:00:00  170.18     163  
3  1982-05-08 00:00:00  182.88     198  
4  1979-11-08 00:00:00  182.88     154

In [167]:
query = """
SELECT COUNT(player_name)  FROM Player
WHERE player_name  LIKE '%smith%'
;
"""
smith = pd.read_sql_query(query, conn)
smith

COUNT(player_name)
0                  18

## QUESTION 5 ANSWER

## QUESTION 6 ANSWER

In [124]:
query="""
SELECT * FROM Player_Attributes;
"""
players = pd.read_sql_query(query, conn)
players.head()

id  player_fifa_api_id  player_api_id                 date  overall_rating  \
0   1              218353         505942  2016-02-18 00:00:00            67.0   
1   2              218353         505942  2015-11-19 00:00:00            67.0   
2   3              218353         505942  2015-09-21 00:00:00            62.0   
3   4              218353         505942  2015-03-20 00:00:00            61.0   
4   5              218353         505942  2007-02-22 00:00:00            61.0   

   potential preferred_foot attacking_work_rate defensive_work_rate  crossing  \
0       71.0          right              medium              medium      49.0   
1       71.0          right              medium              medium      49.0   
2       66.0          right              medium              medium      49.0   
3       65.0          right              medium              medium      48.0   
4       65.0          right              medium              medium      48.0   

   finishing  heading_accuracy  short_passing  volleys  dribbling  curve  \
0       44.0              71.0           61.0     44.0       51.0   45.0   
1       44.0              71.0           61.0     44.0       51.0   45.0   
2       44.0              71.0           61.0     44.0       51.0   45.0   
3       43.0              70.0           60.0     43.0       50.0   44.0   
4       43.0              70.0           60.0     43.0       50.0   44.0   

   free_kick_accuracy  long_passing  ball_control  acceleration  sprint_speed  \
0                39.0          64.0          49.0          60.0          64.0   
1                39.0          64.0          49.0          60.0          64.0   
2                39.0          64.0          49.0          60.0          64.0   
3                38.0          63.0          48.0          60.0          64.0   
4                38.0          63.0          48.0          60.0          64.0   

   agility  reactions  balance  shot_power  jumping  stamina  strength  \
0     59.0       47.0     65.0        55.0     58.0     54.0      76.0   
1     59.0       47.0     65.0        55.0     58.0     54.0      76.0   
2     59.0       47.0     65.0        55.0     58.0     54.0      76.0   
3     59.0       46.0     65.0        54.0     58.0     54.0      76.0   
4     59.0       46.0     65.0        54.0     58.0     54.0      76.0   

   long_shots  aggression  interceptions  positioning  vision  penalties  \
0        35.0        71.0           70.0         45.0    54.0       48.0   
1        35.0        71.0           70.0         45.0    54.0       48.0   
2        35.0        63.0           41.0         45.0    54.0       48.0   
3        34.0        62.0           40.0         44.0    53.0       47.0   
4        34.0        62.0           40.0         44.0    53.0       47.0   

   marking  standing_tackle  sliding_tackle  gk_diving  gk_handling  \
0     65.0             69.0            69.0        6.0         11.0   
1     65.0             69.0            69.0        6.0         11.0   
2     65.0             66.0            69.0        6.0         11.0   
3     62.0             63.0            66.0        5.0         10.0   
4     62.0             63.0            66.0        5.0         10.0   

   gk_kicking  gk_positioning  gk_reflexes  
0        10.0             8.0          8.0  
1        10.0             8.0          8.0  
2        10.0             8.0          8.0  
3         9.0             7.0          7.0  
4         9.0             7.0          7.0

In [148]:
query="""
SELECT COUNT(DISTINCT player_api_id) as num_right_footers
FROM Player_Attributes
WHERE preferred_foot = 'right'

"""
right_footers = pd.read_sql_query(query, conn)
right_footers

num_right_footers
0               8979

In [136]:
query="""
SELECT COUNT(DISTINCT player_api_id) as num_left_footers
FROM Player_Attributes
WHERE preferred_foot = 'left'

"""
left_footers = pd.read_sql_query(query, conn)
left_footers

num_left_footers
0              3202

In [149]:
right_footers_percent = num_right_footers[1:]/(num_right_footers+num_left_footers)
right_footers_percent

NameError: name 'num_right_footers' is not defined